Sources/ZcashLightClientKit/Model/Proposal.swift

```
@@ -8,11 +8,26 @@
 import Foundation
 
 /// A data structure that describes a series of transactions to be created.
-public struct Proposal {
+public struct Proposal: Equatable {
     let inner: FfiProposal
 
     /// Returns the total fee to be paid across all proposed transactions, in zatoshis.
     public func totalFeeRequired() -> Zatoshi {
-        return Zatoshi(Int64(inner.balance.feeRequired))
+        Zatoshi(Int64(inner.balance.feeRequired))
+    }
+}
+
+public extension Proposal {
```

I would advise not to expose this publicly.

---

If you have a suggestion for how to avoid exposing this publicly, I'm all ears. The constraint here is Zashi iOS using TCA, which needs to be able to construct mock data to be returned from its mock implementation of the `Synchronizer` protocol for use in e.g. preview of UIs (so we can't just return something that would be an error case).

---

NOTE: I don't know which things you've tried and which ones you haven't, but there are a few things I can thing from the top of my head. I promise I'll look this deeper shortly. I might be still lacking context and this could be somewhat inaccurate or insufficient. 


### A. "Uncle bob's Clean code" approach
Treat the whole SDK as an external dependency under the principle of Bob Martin's clean code architecture. Look at the problem as you were Unstoppable or EDGE and you don't control the dependency. Wrap around it and have a shim that you Do control and use that instead. 

### B. Pointfree.co / TCA approach. 
Make proposal on the SDK Protocol. The SDK returns an implementation of it. Then use a protocol witness or a mock implementation on Zashi to provide the mocked version that you need. 

### A and B
Make a protocol on Zashi  for what the app expects of a proposal. This propocol contains a throwing `toProposal()` function. 
Make an Adapter of the SDK's Proposal type that conforms to Zashi's. 
Make a Mock implementation for testing on Zashi that throws whenever `toProposal()` is invoked.